### PR TITLE
Build with scala 2.12.2 and oraclejdk8

### PR DESCRIPTION
**Pull request by raboof**
Which are the currently common versions for Scala development

### CODE REVIEW

File: lib/travis/build/script/scala.rb [hunk 1]
```diff
@@ -6,8 +6,8 @@ class Script
       class Scala < Jvm
 
         DEFAULTS = {
-          scala: '2.10.4',
-          jdk:   'default'
+          scala: '2.12.1',
+          jdk:   'oraclejdk8'
```
Comment (BanzaiMan) [inline lib/travis/build/script/scala.rb]: Please remove this for now.

Comment (raboof) [inline lib/travis/build/script/scala.rb]: I'm not sure I understand what exactly you want me to remove - do you mean you want the JDK version to remain on `default`? Scala 2.12 actually requires at least JDK 8 (http://scala-lang.org/news/2.12-roadmap).

File: lib/travis/build/script/scala.rb [hunk 2]
```diff
@@ -6,8 +6,8 @@ class Script
       class Scala < Jvm
 
         DEFAULTS = {
-          scala: '2.10.4',
-          jdk:   'default'
+          scala: '2.12.2',
+          jdk:   'oraclejdk8'
```
Comment (BanzaiMan) [inline lib/travis/build/script/scala.rb]: Please drop this.

Comment (raboof) [inline lib/travis/build/script/scala.rb]: I'm not sure I understand what exactly you want me to drop/remove - do you mean you want the JDK version to remain on default? Scala 2.12 actually requires at least JDK 8 (http://scala-lang.org/news/2.12-roadmap).

Comment (BanzaiMan) [inline lib/travis/build/script/scala.rb]: Sorry, I did not know about this requirement.

In this case, then, we will postpone merging this until we can move to Java 8 for the default JVM version.

Comment (raboof) [inline lib/travis/build/script/scala.rb]: Makes sense! That'd be when trusty images become the default, right? (https://blog.travis-ci.com/2015-10-14-opening-up-ubuntu-trusty-beta/)

Comment (BanzaiMan) [inline lib/travis/build/script/scala.rb]: Yes, I'd say so.

Comment (BanzaiMan) [inline lib/travis/build/script/scala.rb]: Yes, I'd say so.

**Comment by meatballhat** [comment]
As far as I can tell, this PR will require some work in order to be merge-ready, assuming it is still desired and valid. I'm closing, but please holler if you are interested in seeing this PR resurrected 💖 !

**Pull request closed without merging**